packages/mui-material-next/src/ButtonGroup/ButtonGroupContext.ts

```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
-import type { ButtonGroupProps } from './ButtonGroup';
+import type { ButtonGroupProps } from './ButtonGroup.types';
 
-interface ButtonGroupContextType {
+export interface ButtonGroupContextType {
```

Comment:
@mui/core I notice we don't really have a strong convention when using `createContext()` – I can find a lot of createContext(null), createContext(undefined) or createContext({}) 😅 https://github.com/search?q=repo%3Amui%2Fmaterial-ui+createContext+path%3A%2F%5Epackages%5C%2F%2F&type=code

Aside from consistency, are there any differences between having every key being possibly undefined like in this file, or have the entire context possibly being undefined (instead of `{}`) like here: https://github.com/mui/material-ui/blob/master/packages/mui-material-next/src/FormControl/FormControlContext.ts#L15-L30 ?

Reply:
@mj12albert The parameter passed in createContext is the default value that useContext() returns if it is not called in a context. React itself actually recommends that you do not pass undefined or null, but that would be a little too complicated.
See: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24509#issuecomment-382213106

I think the way it is, it should be the nicest way, since the buttons outside a ButtonGroup are also used and so the context within the button acts as if no parameters were used.

Reply:
I don't have a strong preference on undefined vs null, but I do think we should follow the semantic of what these two values mean - not defined, vs defined as a null(none) value. Also, we should be consistent, to avoid surprises. I would avoid empty object as initialization, it's not helping anyhow. There are interesting takes don't he default values for the context here: https://kentcdodds.com/blog/how-to-use-react-context-effectively

Reply:
I see it the same way in principle, but since we are dealing here with a context that is partly used outside a provider, I would do it differently at this point, as we avoid unnecessary checks in the button component, since the context may well be used outside the provider.